pkgs/servers/code-server/default.nix

```
@@ -0,0 +1,139 @@
+    outputHash = {
+      x86_64-linux = "197v9660yqhvazalyhc87ciy6q2ii8xxi5bc5y96bck2w0d2y8if";
+      aarch64-linux = "LiIvGuBismWSL2yV2DuKUWDjIzuIQU/VVxtiD4xJ+6Q=";
+    }.${system} or (throw "Unsupported system ${system}");
+  };
+
+  # Extract the Node.js source code which is used to compile packages with
+  # native bindings
+  nodeSources = runCommand "node-sources" {} ''
+    tar --no-same-owner --no-same-permissions -xf ${nodejs.src}
+    mv node-* $out
+  '';
+
+  nativeBuildInputs = [ nodejs yarn python pkgconfig zip makeWrapper git rsync ];
+  buildInputs = [ libsecret xorg.libX11 xorg.libxkbfile ];
+
+  patchPhase = ''
+    export HOME=$PWD
+
+    # apply code-server vscode patches
+    yarn vscode:patch
+
+    # allow offline install for vscode
+    substituteInPlace lib/vscode/build/npm/postinstall.js --replace '--ignore-optional' '--offline'
+
+    # remove all built-in extensions, as these are 3rd party extensions that gets
+    # downloaded from vscode marketplace
+    echo '[]' > lib/vscode/build/builtInExtensions.json
+
+    # patch commit as autodetection does not work if we don't have .git
+    sed -i '/const commit/c\const commit = "${commit}";' ci/build.ts
+  '';
+
+  configurePhase = ''
+    # set offline mirror to yarn cache we created in previous steps
+    yarn --offline config set yarn-offline-mirror "${yarnCache}"
+
+    npm config set nodedir "${nodeSources}"
+  '';
+
+  buildPhase = ''
+    yarn install --frozen-lockfile --offline --no-progress --non-interactive
+
+    # install without running scripts, for all required packages that needs patching
+    for d in lib/vscode lib/vscode/build; do
+      yarn install --cwd $d --frozen-lockfile --offline --no-progress --non-interactive --ignore-scripts
+    done
+
+    # put ripgrep binary into bin folder, so postinstall does not try to download it
+    find -name vscode-ripgrep -type d \
+      -execdir mkdir -p {}/bin \; \
+      -execdir ln -s ${ripgrep}/bin/rg {}/bin/rg \;
+
+    # patch shebangs of everything, also cached files, as otherwise postinstall
+    # will not be able to find /usr/bin/env, as it does not exists in sandbox
+    patchShebangs .
+
+    (
+      cd lib/vscode
+
+      # rebuild binaries, we use npm here, as yarn does not provider alternative
+      npm rebuild --update-binary
+
+      # run postinstall scripts, which eventually do yarn install on all additional requirements
+      yarn postinstall --offline --frozen-lockfile
+    )
+
+    yarn build
+  '';
+
+  installPhase = ''
+    mkdir -p $out/libexec/code-server $out/bin
+    cp -R -T build $out/libexec/code-server
+
+    makeWrapper "${nodejs}/bin/node" "$out/bin/code-server" \
+      --add-flags "$out/libexec/code-server/out/node/entry.js"
+  '';
+
+  passthru = {
+    prefetchYarnCache = overrideDerivation yarnCache (d: {
+      outputHash = "0000000000000000000000000000000000000000000000000000000000000000";
```

BTW `stdenv.lib.fakeSha256` exists with this same value, if you want to avoid embedding a long string of zeroes and maybe make it more semantically meaningful.

```suggestion
      outputHash = stdenv.lib.fakeSha256;
```

---

Done